File: languages/php/Bloock/Entity/HostedLoader.php
<?php

namespace Bloock\Entity;

use Bloock\DataAvailabilityType;

class HostedLoader implements Loader
{
    private int $type;
    private LoaderArgs $args;

    public function __construct(string $id) {
        $this->type = DataAvailabilityType::HOSTED;
        $this->args = new LoaderArgs($id);
    }

    public function toProto(): \Bloock\Loader {
        $p = new \Bloock\Loader();
        $p->setType($this->type);
        $p->setArgs($this->args->toProto());
        return $p;
    }
}

File: languages/java/src/main/java/com/bloock/sdk/entity/availability/HostedLoader.java
package com.bloock.sdk.entity.availability;

import com.bloock.sdk.bridge.proto.BloockAvailabilityEntities;
import com.bloock.sdk.bridge.proto.BloockAvailabilityEntities.DataAvailabilityType;

/**
 * Represents a loader for hosted data availability.
 */
public class HostedLoader implements Loader {
  DataAvailabilityType type;
  LoaderArgs args;

  /**
   * Constructs a HostedLoader object with the specified parameters.
   * 
   * @param id
   */
  public HostedLoader(String id) {
    type = DataAvailabilityType.HOSTED;
    args = new LoaderArgs(id);
  }

  public BloockAvailabilityEntities.Loader toProto() {
    return BloockAvailabilityEntities.Loader.newBuilder()
        .setType(this.type)
        .setArgs(this.args.toProto())
        .build();
  }
}
